app/assets/javascripts/admin/addon/controllers/admin-user-index.js

```diff
@@ -170,6 +177,37 @@ export default class AdminUserIndexController extends Controller {
     return lastPayload.split("&");
   }
 
+  @action
+  triggerUserExport() {
+    this.dialog.yesNoConfirm({
+      message: i18n("admin.user.exports.download.confirm"),
+      didConfirm: async () => {
+        this.userExportReloading = true;
+        const currentExportUri = this.userExport?.uri;
+        try {
+          await exportEntity("user_archive", { export_user_id: this.model.id });
+          this.dialog.alert(i18n("admin.user.exports.download.success"));
+
+          const interval = setInterval(async () => {
+            this.userExport = await UserExport.findLatest(this.model.id);
+            if (this.userExport.uri !== currentExportUri) {
+              this.userExportReloading = false;
+              clearInterval(interval);
+            }
```

Comment:
This feels like a bit of a hacky way to check a model for updates. Is there a better way to set it up?

Reply:
That's much nicer, thanks for the pointer!